src/app.rs

```diff
@@ -484,14 +485,26 @@ where
             info!("Rustc git commit: {}", hash);
         }
 
+        let thread_count :usize = env::var("AMETHYST_NUM_THREADS")
+            .as_ref()
+            .map(|s| s.as_str().parse()
+                    .expect("AMETHYST_NUM_THREADS was provided but is not a valid number!"))
+            .unwrap_or(0);
```

Comment:
instead of `unwrap_or(0)`, how about leaving this as `Option<usize>`, and then use `if let Some(thread_count) = thread_count` below?

That way we don't have `0` as a sentinel value, and if the user really wanted to specify 0, then they can do it

Reply:
Ah, yes, I didn't wrap my head about if-let yet, but this makes sense so will adjust accordingly. Thank you kindly!